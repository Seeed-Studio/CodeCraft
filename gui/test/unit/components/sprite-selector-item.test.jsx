import React from 'react';
import {mountWithIntl, shallowWithIntl, componentWithIntl} from '../../helpers/intl-helpers.jsx';
import SpriteSelectorItemComponent from '../../../src/components/sprite-selector-item/sprite-selector-item';
import CloseButton from '../../../src/components/close-button/close-button';

describe('SpriteSelectorItemComponent', () => {
    let className;
    let costumeURL;
    let name;
    let onClick;
    let onDeleteButtonClick;
    let selected;
    let number;
    let details;

    // Wrap this in a function so it gets test specific states and can be reused.
    const getComponent = function () {
        return (
            <SpriteSelectorItemComponent
                className={className}
                costumeURL={costumeURL}
                details={details}
                name={name}
                number={number}
                selected={selected}
                onClick={onClick}
                onDeleteButtonClick={onDeleteButtonClick}
            />
        );
    };

    beforeEach(() => {
        className = 'ponies';
        costumeURL = 'https://scratch.mit.edu/foo/bar/pony';
        name = 'Pony sprite';
        onClick = jest.fn();
        onDeleteButtonClick = jest.fn();
        selected = true;
        // Reset to undefined since they are optional props
        number = undefined; // eslint-disable-line no-undefined
        details = undefined; // eslint-disable-line no-undefined
    });

    test('matches snapshot when selected', () => {
        const component = componentWithIntl(getComponent());
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('matches snapshot when given a number and details to show', () => {
        number = 5;
        details = '480 x 360';
        const component = componentWithIntl(getComponent());
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('does not have a close box when not selected', () => {
        selected = false;
        const wrapper = shallowWithIntl(getComponent());
        expect(wrapper.find(CloseButton).exists()).toBe(false);
    });

    test('triggers callback when Box component is clicked', () => {
        // Use `mount` here because of the way ContextMenuTrigger consumes onClick
        const wrapper = mountWithIntl(getComponent());
        wrapper.simulate('click');
        expect(onClick).toHaveBeenCalled();
    });

    test('triggers callback when CloseButton component is clicked', () => {
        const wrapper = shallowWithIntl(getComponent());
        wrapper.find(CloseButton).simulate('click');
        expect(onDeleteButtonClick).toHaveBeenCalled();
    });

    test('it has a context menu with delete menu item and callback', () => {
        const wrapper = mountWithIntl(getComponent());
        const contextMenu = wrapper.find('ContextMenu');
        expect(contextMenu.exists()).toBe(true);

        contextMenu.find('[children="delete"]').simulate('click');
        expect(onDeleteButtonClick).toHaveBeenCalled();
    });
});
