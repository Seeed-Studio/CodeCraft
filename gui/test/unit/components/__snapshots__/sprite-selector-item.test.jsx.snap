// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpriteSelectorItemComponent matches snapshot when given a number and details to show 1`] = `
<div
  className="react-contextmenu-wrapper ponies undefined"
  onClick={[Function]}
  onContextMenu={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={undefined}
  onMouseLeave={undefined}
  onMouseOut={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
>
  <div
    aria-label="Close"
    className=""
    onClick={[Function]}
    role="button"
    tabIndex="0"
  >
    <img
      className=""
      src="test-file-stub"
    />
  </div>
  <div
    className={undefined}
  >
    5
  </div>
  <img
    className={undefined}
    src="https://scratch.mit.edu/foo/bar/pony"
  />
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      Pony sprite
    </div>
    <div
      className={undefined}
    >
      480 x 360
    </div>
  </div>
  <nav
    className="react-contextmenu"
    onContextMenu={[Function]}
    onMouseLeave={[Function]}
    role="menu"
    style={
      Object {
        "opacity": 0,
        "pointerEvents": "none",
        "position": "fixed",
      }
    }
    tabIndex="-1"
  >
    <div
      aria-disabled="false"
      aria-orientation={null}
      className="react-contextmenu-item"
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      role="menuitem"
      tabIndex="-1"
    >
      <span>
        delete
      </span>
    </div>
  </nav>
</div>
`;

exports[`SpriteSelectorItemComponent matches snapshot when selected 1`] = `
<div
  className="react-contextmenu-wrapper ponies undefined"
  onClick={[Function]}
  onContextMenu={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={undefined}
  onMouseLeave={undefined}
  onMouseOut={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
>
  <div
    aria-label="Close"
    className=""
    onClick={[Function]}
    role="button"
    tabIndex="0"
  >
    <img
      className=""
      src="test-file-stub"
    />
  </div>
  <img
    className={undefined}
    src="https://scratch.mit.edu/foo/bar/pony"
  />
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      Pony sprite
    </div>
  </div>
  <nav
    className="react-contextmenu"
    onContextMenu={[Function]}
    onMouseLeave={[Function]}
    role="menu"
    style={
      Object {
        "opacity": 0,
        "pointerEvents": "none",
        "position": "fixed",
      }
    }
    tabIndex="-1"
  >
    <div
      aria-disabled="false"
      aria-orientation={null}
      className="react-contextmenu-item"
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      role="menuitem"
      tabIndex="-1"
    >
      <span>
        delete
      </span>
    </div>
  </nav>
</div>
`;
